src/github.com/matrix-org/go-neb/database/db.go

```diff
@@ -17,15 +17,15 @@ type ServiceDB struct {
 // A single global instance of the service DB.
 // XXX: I can't think of any way of doing this without one without creating
 //      cyclical dependencies somewhere -- Kegan
```

Comment:
(I realise this isn't actually part of the diff. But could you elaborate on what you wanted to do and why it resulted in cycles?)